### ISSUE TITLE
(FunctionClauseError) no function clause matching in Phx2Ban.Rules.KnownMaliciousIp.check_request/2

### ISSUE THREAD

**Issue by jaeyson** [ezsuite/community#3]
Hi! This was taken from Sentry:

```bash
GenServer #PID<0.2947.0> terminating
** (stop) exited in: GenServer.call(Phx2Ban.Rules.KnownMaliciousIp, {:valid_request?, {17777, 33333, 1, 1, 1, 11, 54444, 0}}, 5000)
    ** (EXIT) an exception was raised:
        ** (FunctionClauseError) no function clause matching in Phx2Ban.Rules.KnownMaliciousIp.check_request/2
            (phx_2_ban 0.2.0) lib/phx_2_ban/rules/known_malicious_ip.ex:67: Phx2Ban.Rules.KnownMaliciousIp.check_request({17777, 33333, 1, 1, 1, 11, 54444, 0}, {:cuckoo_filter, #Reference<0.425678620.1433272321.3030>, {:spinlock, #Reference<0.425678620.1433272321.3030>, 1000000}, 32768, 2147483647, 4, 16, 100, #Function<0.131942963/1 in Phx2Ban.Rules.KnownMaliciousIp.hash>})
            (phx_2_ban 0.2.0) lib/phx_2_ban/rules/known_malicious_ip.ex:54: Phx2Ban.Rules.KnownMaliciousIp.handle_call/3
            (stdlib 6.1.2) gen_server.erl:2381: :gen_server.try_handle_call/4
            (stdlib 6.1.2) gen_server.erl:2410: :gen_server.handle_msg/6
            (stdlib 6.1.2) proc_lib.erl:329: :proc_lib.init_p_do_apply/3
    (elixir 1.17.3) lib/gen_server.ex:1128: GenServer.call/3
    (phx_2_ban 0.2.0) lib/phx_2_ban/analyzer.ex:134: anonymous fn/2 in Phx2Ban.Analyzer.maybe_blocklist_ip/2
    (elixir 1.17.3) lib/enum.ex:4343: Enum.find_value_list/3
    (phx_2_ban 0.2.0) lib/phx_2_ban/analyzer.ex:91: Phx2Ban.Analyzer.handle_conn_data/2
    (phx_2_ban 0.2.0) lib/phx_2_ban/analyzer.ex:61: Phx2Ban.Analyzer.handle_cast/2
    (stdlib 6.1.2) gen_server.erl:2371: :gen_server.try_handle_cast/3
    (stdlib 6.1.2) gen_server.erl:2433: :gen_server.handle_msg/6
    (stdlib 6.1.2) proc_lib.erl:329: :proc_lib.init_p_do_apply/3
Last message: {:"$gen_cast", {:analyze_request, %Phx2Ban.ConnData{remote_ip: {17777, 33333, 1, 1, 1, 11, 54444, 0}, method: "GET", request_path: "/", req_headers: [{"accept", "text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.7"}, {"accept-encoding", "gzip, deflate, br"}, {"accept-language", "en-US,ja-JP;q=0.9,ja;q=0.8,en;q=0.7"}, {"fly-client-ip", "1201:7898:1:7ba::ff"}, {"fly-forwarded-port", "443"}, {"fly-forwarded-proto", "https"}, {"fly-forwarded-ssl", "on"}, {"fly-region", "lax"}, {"fly-request-id", "01112222223333333337555449-lax"}, {"host", "[domain.co](http://domain.co/)"}, {"priority", "u=4, i"}, {"purpose", "prefetch"}, {"referer", "https://www.google.com/"}, {"sec-ch-ua", "\"Google Chrome\";v=\"131\", \"Chromium\";v=\"131\", \"Not_A Brand\";v=\"24\""}, {"sec-ch-ua-mobile", "?1"}, {"sec-ch-ua-platform", "\"Android\""}, {"sec-fetch-dest", "empty"}, {"sec-fetch-mode", "no-cors"}, {"sec-fetch-site", "none"}, {"sec-purpose", "prefetch;anonymous-client-ip"}, {"user-agent", "Mozilla/5.0 (Linux; Android 10; K) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/[131.0.0.0](http://131.0.0.0/) Mobile Safari/537.36"}, {"via", "2 [fly.io](http://fly.io/)"}, {"x-forwarded-for", "1201:7898:1:7ba::ff, 1201:7898:1:7ba::ff"}, {"x-forwarded-port", "443"}, {"x-forwarded-proto", "https"}, {"x-forwarded-ssl", "on"}, {"x-request-start", "t=1133222333377733"}], status: nil, host: "[domain.co](http://domain.co/)", req_route_exists?: false}}}
```

environment = prod
level = error
os = linux 5.15.98
os.name = linux
runtime = elixir 1.17.3 (compiled with Erlang/OTP 27)
runtime.name = elixir
server_name = eeee8669975555

**Comment by jaeyson** [comment]
```bash
FunctionClauseError: no function clause matching in Phx2Ban.Rules.KnownMaliciousIp.check_request/2
  File "lib/phx_2_ban/rules/known_malicious_ip.ex", line 67, in Phx2Ban.Rules.KnownMaliciousIp.check_request/2
  File "lib/phx_2_ban/rules/known_malicious_ip.ex", line 54, in Phx2Ban.Rules.KnownMaliciousIp.handle_call/3
  File "gen_server.erl", line 2381, in :gen_server.try_handle_call/4
  File "gen_server.erl", line 2410, in :gen_server.handle_msg/6
  File "proc_lib.erl", line 329, in :proc_lib.init_p_do_apply/3
````
<img width="676" alt="Screenshot 2024-11-29 at 19 45 31" src="https://github.com/user-attachments/assets/7c42af0b-962c-409d-84dc-a26d48edb1cd">

**Comment by jaeyson** [comment]
I disabled the "En/Disable Firewall" button for now

**Comment by PJUllrich** [comment]
hey @jaeyson thanks for the bug report. I'm working on it. The problem is that we don't support IPv6 (yet). I'll add a quickfix for this one.

**Comment by PJUllrich** [comment]
@jaeyson I just released `v0.2.1` which should fix this. could you please bump the version and try again? thank you for your patience!
